### PR TITLE
New DNS Seeders for DOGE

### PR DESCRIPTION
**Pull request by carsenk**
Courtesy of https://denarius.io Denarius (D) community we are providing Dogecoin some DNS peer seeders, this PR includes the seeders hard coded in

```
veryseed.denarius.pro
muchseed.denarius.pro
suchseed.denarius.pro (WIP still but should be online)
```

### CODE REVIEW

**Comment by CircuitBreaker88** [comment]
Nice!

**Comment by rnicoll** [comment]
How did you test these work? Same question seed2. So:

```
jrn@testnet-node:~$ dig @ns2.multidoge.org seed2.multidoge.org

; <<>> DiG 9.11.3-1ubuntu1.13-Ubuntu <<>> @ns2.multidoge.org seed2.multidoge.org
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 11463
;; flags: qr aa rd; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;seed2.multidoge.org.           IN      A

;; ANSWER SECTION:
seed2.multidoge.org.    600     IN      A       31.165.17.142
seed2.multidoge.org.    600     IN      A       91.86.219.71
seed2.multidoge.org.    600     IN      A       73.167.72.250
seed2.multidoge.org.    600     IN      A       71.161.223.193

;; Query time: 81 msec
;; SERVER: 63.34.247.220#53(63.34.247.220)
;; WHEN: Sat Jan 30 13:42:36 CET 2021
;; MSG SIZE  rcvd: 177
```

That's what I'd expect. On veryseed at least:

```
$ dig @veryseed.denarius.pro veryseed.denarius.pro

; <<>> DiG 9.11.3-1ubuntu1.13-Ubuntu <<>> @veryseed.denarius.pro veryseed.denarius.pro
; (10 servers found)
;; global options: +cmd
;; connection timed out; no servers could be reached
```

Fairly certain the nameserver and address can't be the same, because then you're trying to resolve the IP of the nameserver by querying itself, and that confuses the hell out of things.

**Comment by carsenk** [comment]
@rnicoll 

```
dig -t A veryseed.denarius.pro

; <<>> DiG 9.16.1-Ubuntu <<>> -t A veryseed.denarius.pro
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 3518
;; flags: qr rd ra; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;veryseed.denarius.pro.        IN    A

;; ANSWER SECTION:
veryseed.denarius.pro.    3024    IN    A    138.68.244.109
veryseed.denarius.pro.    3024    IN    A    93.114.128.120
veryseed.denarius.pro.    3024    IN    A    87.100.205.39
veryseed.denarius.pro.    3024    IN    A    73.221.48.43
veryseed.denarius.pro.    3024    IN    A    78.56.87.65
veryseed.denarius.pro.    3024    IN    A    206.172.73.120

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Sat Jan 30 09:29:24 PST 2021
;; MSG SIZE  rcvd: 14
```

```
dig -t NS veryseed.denarius.pro

; <<>> DiG 9.16.1-Ubuntu <<>> -t NS veryseed.denarius.pro
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 64243
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;veryseed.denarius.pro.        IN    NS

;; ANSWER SECTION:
veryseed.denarius.pro.    6894    IN    NS    verydoge.denarius.pro.

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Sat Jan 30 09:28:07 PST 2021
;; MSG SIZE  rcvd: 73
```
verydoge.denarius.pro hosts the name service for veryseed.denarius.pro

File: src/chainparams.cpp [hunk 1]
```diff
@@ -162,7 +162,10 @@ class CMainParams : public CChainParams {
 
         // Note that of those with the service bits flag, most only support a subset of possible options
         vSeeds.push_back(CDNSSeedData("multidoge.org", "seed.multidoge.org", true));
-        vSeeds.push_back(CDNSSeedData("multidoge.org", "seed2.multidoge.org"));
+        vSeeds.push_back(CDNSSeedData("multidoge.org", "seed2.multidoge.org")); // Seeders are dead
```
Comment (patricklodder) [inline src/chainparams.cpp:165]: according to my checks over the past 24h, this seeder is not dead. The community may want to put some downtime detection on individual seeders though.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -162,7 +162,10 @@ class CMainParams : public CChainParams {
...
+        vSeeds.push_back(CDNSSeedData("veryseed.denarius.pro", "veryseed.denarius.pro"));
+        vSeeds.push_back(CDNSSeedData("muchseed.denarius.pro", "muchseed.denarius.pro"));
+        vSeeds.push_back(CDNSSeedData("suchseed.denarius.pro", "suchseed.denarius.pro")); //New working seeders Courtesy of Denarius (D) denarius.io Such D Much Wow
```
Comment (patricklodder) [inline src/chainparams.cpp:168]: Can we remove the comment?

**Review by patricklodder — CHANGES_REQUESTED** [review]
Please remove unnecessary comments.

File: src/chainparams.cpp (hunk 2 above)
Comment (carsenk) [reply src/chainparams.cpp]: Removed all :)

**Comment by carsenk** [comment]
Removed Comments. To the moon!

**Comment by patricklodder** [comment]
- [x] Stability of DNS server confirmed over the past 48h (will still continue)
- [x] Crawler does not misbehave 
   
Per this: https://github.com/dogecoin/dogecoin/blob/master/doc/dnsseed-policy.md
- [ ] Could you please confirm that you comply to "4. Any logging of DNS queries should be only that which is necessary for the operation of the service or urgent health of the Dogecoin network and **must not be retained longer than necessary nor disclosed to any third party**."?
- [ ] Per "6. Operators are encouraged, but not required, to publicly document the details of their operating practices." Do you have such documentation and/or is your crawler open-source?
- [ ] 7. Please provide an email address

Once that is done, I'm good.

**Comment by buzzkillb** [comment]
> 
> 
>     * [x]  Stability of DNS server confirmed over the past 48h (will still continue)
> 
>     * [x]  Crawler does not misbehave
> 
> 
> Per this: https://github.com/dogecoin/dogecoin/blob/master/doc/dnsseed-policy.md
> 
>     * [ ]  Could you please confirm that you comply to "4. Any logging of DNS queries should be only that which is necessary for the operation of the service or urgent health of the Dogecoin network and **must not be retained longer than necessary nor disclosed to any third party**."?
> 
>     * [ ]  Per "6. Operators are encouraged, but not required, to publicly document the details of their operating practices." Do you have such documentation and/or is your crawler open-source?
> 
>     * [ ]  7. Please provide an email address
> 
> 
> Once that is done, I'm good.

- Logging is whatever the stats dump is in the folder during crawling to run the seeder. 3rd parties will never be given access to the servers or share the info out.
- Crawler is open source fork of Sipa but easier to configure, https://github.com/team-exor/generic-seeder
- buzz@denarius.io

**Comment by carsenk** [comment]
> * Logging is whatever the stats dump is in the folder during crawling to run the seeder. 3rd parties will never be given access to the servers or share the info out.
> * Crawler is open source fork of Sipa but easier to configure, https://github.com/team-exor/generic-seeder
> * [buzz@denarius.io](mailto:buzz@denarius.io)

^^^

**Review by mxaddict — APPROVED** [review]

**Comment by arulnick** [comment]
I'm just curious why there isn't a requirement for use of OpenSSL/SHA 256 connections?  Mostly just wondering/devils advocate. Sorry to hijack just a question that came to my mind.

**Comment by patricklodder** [comment]
> I'm just curious why there isn't a requirement for use of OpenSSL/SHA 256 connections? Mostly just wondering/devils advocate. Sorry to hijack just a question that came to my mind.

On DNS? Because I am not at all sure that with our `getaddrinfo()`-dependent implementation we can actually do DNS over TLS. Is worth looking into, but not the 1.14.3 release target. Be my guest to do the research on how it could be implemented and we can look at that in a subsequent release.

**Review by patricklodder — APPROVED** [review]
ACK on the code.
ACK on compliance.

**Review by langerhans — APPROVED** [review]
ACK

**Comment by bruidbarrett** [comment]
@langerhans I have a very important question regarding the langerhans dogecoin wallet, I messaged you on twitter about it and if there is a good way to contact you please let me know or email me, bruidbarrett@gmail.com. it is VERY IMPORTANT.

**Comment by langerhans** [comment]
@bruidbarrett Ping me on Discord, but believe, it's currently VERY IMPORTANT for everyone...

**Comment by bruidbarrett** [comment]
What is your discord?

**Comment by bruidbarrett** [comment]
@langerhans what is your discord? Sorry I am sure you are busy man!

**Comment by langerhans** [comment]
I'm on the Dogecoin discord. langerhans#2409. Ping me there.

**Comment by rnicoll** [comment]
Thank you!

**Pull request merged**